src/ReactiveSocket.cpp

```diff
@@ -59,9 +62,14 @@ ReactiveSocket::ReactiveSocket(
       keepaliveTimer_(std::move(keepaliveTimer)),
       resumeSocketListener_(resumeListener) {}
 
+folly::Executor& ReactiveSocket::defaultExecutor() {
```

Comment:
It seems like in this case you have to make the calls from the correct thread to have correctness.  Is there a folly executor decorator or something similar that can do those checks for us?